hudi-spark-datasource/hudi-spark3-common/src/main/scala/org/apache/spark/sql/connector/catalog/HoodieCatalogAndIdentifier.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog
+
+import org.apache.spark.sql.internal.{SQLConf, StaticSQLConf}
+
+object HoodieCatalogAndIdentifier {
+
+  import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.MultipartIdentifierHelper
+
+  private val globalTempDB = SQLConf.get.getConf(StaticSQLConf.GLOBAL_TEMP_DATABASE)
+
+  def parse(catalogManager: CatalogManager, nameParts: Seq[String])
+      : Option[(CatalogPlugin, Identifier)] = {
+    assert(nameParts.nonEmpty)
+    val currentCatalog = catalogManager.currentCatalog
+    if (nameParts.length == 1) {
+      Some((currentCatalog, Identifier.of(catalogManager.currentNamespace, nameParts.head)))
+    } else if (nameParts.head.equalsIgnoreCase(globalTempDB)) {
+      // Conceptually global temp views are in a special reserved catalog. However, the v2 catalog
```

Comment:
Was this borrowed from somewhere?

Reply:
yep, from `org.apache.spark.sql.connector.catalog.LookupCatalog.CatalogAndIdentifier`.
But `LookupCatalog` is a trait, and `CatalogAndIdentifier` is its inner object. If i make `HoodieSpark3CatalystPlanUtils` extend to `LookupCatalog`, i have to override the `catalogManager` that is not initializeed in `LookupCatalog`, further i have to pass a `SparkSession` to here, further... It will affect the definition of `HoodieCatalystPlansUtils` and `SparkAdapter` and `SparkAdapterSupport` and other places where use them.
So i extract the logical here, use it as a method call.

Reply:
There's actually no need for that,  you can simply do following:

```
  def resolve(spark: SparkSession, relation: UnresolvedRelation) = {
    import spark.sessionState.analyzer.CatalogAndIdentifier

    val CatalogAndIdentifier(plugin, ident) = relation.multipartIdentifier
    
    // ...
  }
```

Let's make sure we limit amount of code we borrow from Spark to exceptional cases when there's just no other way around it

Reply:
yes, you're right. I don't know `Analyzer` has the `CatalogAndIdentifier` object before.